velox/expression/CastExpr.cpp

```diff
@@ -134,6 +123,29 @@ void applyIntToDecimalCastKernel(
   });
 }
 
+template <typename TInput>
+VectorPtr applyDecimalToDoubleCast(
+    const SelectivityVector& rows,
+    const BaseVector& input,
+    exec::EvalCtx& context,
+    const TypePtr& fromType) {
+  VectorPtr result;
+  context.ensureWritable(rows, DOUBLE(), result);
+  (*result).clearNulls(rows);
+  auto sourceVector = input.as<SimpleVector<TInput>>();
+  auto resultBuffer =
+      result->asUnchecked<FlatVector<double>>()->mutableRawValues();
+  auto toPrecisionScale = getDecimalPrecisionScale(*fromType);
+  context.applyToSelectedNoThrow(rows, [&](int row) {
+    auto output =
+        util::Converter<CppToType<double>::typeKind, void, false>::cast(
+            input.as<SimpleVector<TInput>>()->valueAt(row));
+    resultBuffer[row] =
+        output / (double)DecimalUtil::kPowersOfTen[toPrecisionScale.second];
```

Comment:
Let's move (double)DecimalUtil::kPowersOfTen[toPrecisionScale.second] outside the loop as well.